hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestClientSideRegionScanner.java

```diff
@@ -179,6 +181,72 @@ public void testContinuesToScanIfHasMore() throws IOException {
     }
   }
 
+  @Test
+  public void testScanMetricsByRegion() throws IOException {
+    Scan scan = new Scan();
+    scan.setScanMetricsEnabled(true);
+    scan.setEnableScanMetricsByRegion(true);
+
+    Configuration copyConf = new Configuration(conf);
+    // Test without providing scan metric at prior and scan metrics by region enabled
```

Comment:
nit: each of these little blocks could be its own test method, and then this test would be a lot nicer to reason about when there are failures.